library/libc/malloc.c

```diff
@@ -17,7 +17,7 @@ static char* brk = &__heap_start;
  * If malloc() finds this region to be too small, it will call _sbrk().
  */
 
-char *_sbrk(int size) {
+char *_sbrk(uint size) {
```

Comment:
The parameter of `sbrk` is int type. You could do `man sbrk` in Linux and see the following definition
```
void *
sbrk(int incr);
```
Indeed, `sbrk` can take a negative number as argument and shrink the heap! :-)